ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -788,6 +833,286 @@ private void migrateToPropertyStore() throws InterruptedException {
+          }
+
+          ConfigAccessor configAccessor =
+              new ConfigAccessor(dataCenterToZkAddress.get(dcName).getZkConnectStrs().get(0));
+
+          // 1. Update cluster config
+          setupCluster(configAccessor);
+
+          // 2. Update instance config for all instances present in each resource
+          Set<String> allInstances = new HashSet<>();
+          for (String resource : resources) {
+            // a. Get list of instances sharing partitions in this resource
+            IdealState idealState = resourceToIdealState.get(resource);
+            Set<String> partitions = idealState.getPartitionSet();
+            Set<String> instances = new HashSet<>();
+            for (String partition : partitions) {
+              instances.addAll(idealState.getInstanceSet(partition));
+            }
+            // b. Update instance config for each instance
+            for (String instance : instances) {
+              setupInstanceConfig(dcName, instance, resource, configAccessor, propertyStoreAdapter);
+            }
+            allInstances.addAll(instances);
+          }
+
+          // 3. No resource config to update for now
+
+          // 4. Validate configs
+          Map<String, Boolean> instanceValidationResultMap =
+              helixAdmin.validateInstancesForWagedRebalance(clusterName, new ArrayList<>(allInstances));
+          if (instanceValidationResultMap.containsValue(Boolean.FALSE)) {
+            throw new IllegalStateException("Instances are not configured correctly for waged rebalancer");
+          }
+          Map<String, Boolean> resourceValidationResultMap =
+              helixAdmin.validateResourcesForWagedRebalance(clusterName, new ArrayList<>(resources));
+          if (resourceValidationResultMap.containsValue(Boolean.FALSE)) {
+            throw new IllegalStateException("Resources are not configured correctly for waged rebalancer");
+          }
+
+          if (!dryRun) {
+            // Before
+            Map<String, Set<String>> replicaPlacementBefore = getReplicaPlacement(helixAdmin, resources);
+
+            // 5. Enter maintenance mode
+            helixAdmin.manuallyEnableMaintenanceMode(clusterName, true, "Migrating to Full auto",
+                Collections.emptyMap());
+
+            // 6. Update ideal state and enable waged rebalancer
+            for (String resource : resources) {
+              IdealState idealState = resourceToIdealState.get(resource);
+              idealState.setRebalanceMode(IdealState.RebalanceMode.FULL_AUTO);
+              // Set resource tag
+              idealState.setInstanceGroupTag(getResourceTag(resource));
+              // Set number of replicas
+              idealState.setReplicas(NUM_REPLICAS);
+              idealState.setMinActiveReplicas(MIN_ACTIVE_REPLICAS);
+              helixAdmin.updateIdealState(clusterName, resource, idealState);
```

Comment:
might as well just use setIdealState now that you have all the fields for the ideal state

Reply:
Sorry, I didn't find setIdealState but found addIdealState. I think would still need configs like stateModel, number of partitions, etc. Due to that, left it as updateIdealState